mod-clans/src/main/java/gg/solarmc/loader/clans/Clan.java

```diff
@@ -0,0 +1,414 @@
+     */
+    public int currentKills() {
+        return clanKills;
+    }
+
+    /**
+     * Returns the deaths at the time of querying (not accurate)
+     * @return deaths
+     */
+    public int currentDeaths() {
+        return clanDeaths;
+    }
+
+    /**
+     * Returns the ass at the time of querying (not accurate)
+     * @return ass
+     */
+    public int currentAssists() {
+        return clanAssists;
+    }
+
+    /**
+     * Gets all current members currently in the clan. Not accurate.
+     * @return all ClanMembers.
+     */
+    public Set<ClanMember> currentMembers() {
+        return this.members;
+    }
+
+    /**
+     * Returns a record of info
+     * @param transaction the tx
+     * @return the record
+     */
+    private ClansClanInfoRecord getInformation(Transaction transaction) {
+        ClansClanInfoRecord rec =  transaction.getProperty(DSLContext.class)
+                .fetchOne(CLANS_CLAN_INFO,CLANS_CLAN_INFO.CLAN_ID.eq(this.clanID));
+
+        assert rec != null : "nullity check failed for record get!";
+
+        return rec;
+    }
+
+    /**
+     * Adds kills to a clan
+     * @param transaction the tx
+     * @param toAdd how much to add
+     * @return accurate amount of kills post transaction
+     */
+    public Integer addKills(Transaction transaction, int toAdd) {
+        ClansClanInfoRecord rec = getInformation(transaction);
+
+        int newValue = rec.getClanKills() + toAdd;
+
+        rec.setClanKills(newValue);
+        rec.store(CLANS_CLAN_INFO.CLAN_KILLS);
+
+        this.clanKills = newValue;
+
+        return newValue;
+    }
+
+    /**
+     * Adds deaths to a gg.solarmc.loader.clans.Clan
+     * @param transaction the tx
+     * @param toAdd amount of deaths to add
+     * @return accurate count of deaths post transaction
+     */
+    public Integer addDeaths(Transaction transaction, int toAdd) {
+        ClansClanInfoRecord rec = getInformation(transaction);
+
+        int newValue = rec.getClanDeaths() + toAdd;
+
+        rec.setClanDeaths(newValue);
+        rec.store(CLANS_CLAN_INFO.CLAN_DEATHS);
+
+        this.clanDeaths = newValue;
+
+        return newValue;
+    }
+
+    /**
+     * Adds assists to a gg.solarmc.loader.clans.Clan.
+     * @param transaction tx
+     * @param toAdd the amount you want to add
+     * @return Accurate assists after transaction
+     */
+    public Integer addAssists(Transaction transaction, int toAdd) {
+        ClansClanInfoRecord rec = getInformation(transaction);
+
+        int newValue = rec.getClanAssists() + toAdd;
+
+        rec.setClanAssists(newValue);
+        rec.store(CLANS_CLAN_INFO.CLAN_ASSISTS);
+
+        this.clanAssists = newValue;
+
+        return newValue;
+    }
+
+    /**
+     * Returns the accurate allied clan to this object.
+     * @param transaction
+     * @return
+     */
+    public Optional<Clan> getAlliedClan(Transaction transaction) {
+        if (this.alliedClan == null) return Optional.empty();
```

Comment:
I think there may be an issue with this method implementation. You're relying on `alliedClan` which is a cached and possibly inaccurate value. Callers of this method will expect an accurate, up-to-date value.

Reply:
Outdated, check genfixes 2